lib/streams/recordStream.js

```diff
@@ -8,6 +9,29 @@ var CleanupStream = require('./cleanupStream');
 var ValidRecordFilterStream = require('./validRecordFilterStream');
 var DocumentStream = require('./documentStream');
 
+/*
+ * Construct a suitable id prefix for a CSV file given
+ * its full filename and the base directory of all OA CSV files.
+ */
+function getIdPrefix(filename, dirPath) {
+  var heirarchy = [];
+  var current_level = filename;
+
+  // traverse the directory structure upward to the OA base directory,
+  // keeping track of directory names along the way
+  while (current_level !== '/') {
```

Comment:
I'm normally not a fan of iterative code like this, but the [split based approach](https://github.com/pelias/openaddresses/blob/use_filename_in_id_alternate/lib/streams/recordStream.js#L17) doesn't read much better in my opinion. I'm open to suggestions to make either method cleaner.